components/eventing-controller/controllers/backend/reconciler.go

```diff
@@ -640,6 +640,8 @@ func (r *Reconciler) CreateOrUpdatePublisherProxy(ctx context.Context, backend e
 	}
 
 	desiredPublisher.ResourceVersion = currentPublisher.ResourceVersion
+	desiredPublisher.Spec.Template.ObjectMeta.Annotations = currentPublisher.Spec.Template.ObjectMeta.Annotations
```

Comment:
This check means, that we accept any annotation that was put on the template. 
Annotations such as the infamous `sidecar.istio.io/inject` must be set correctly as they could break our functionality.

Lets ignore specific annotations instead of this general approach

Reply:
Done.